src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp

```diff
@@ -856,48 +880,147 @@ void BrgemmEmitter::initBrgemm(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>&
     brgKernel.reset(brgKernel_);
 }
 
+size_t BrgemmEmitter::aux_gprs_count() const {
+    return m_N_blk_loop + m_K_blk_loop;
+}
+
+void BrgemmEmitter::emit_N_blocking_loops(size_t k_kernel_id,
+                                          const Xbyak::Reg64& input_0, const Xbyak::Reg64& input_1,
+                                          const Xbyak::Reg64& input_2, const Xbyak::Reg64& output_0,
+                                          const Xbyak::Reg64& work_amount_N) const {
+    auto emit_and_shift_pointers = [&](size_t kernel_idx) {
+        const auto& brgemmCtx = m_brgCtxs[kernel_idx];
+        emit_brgemm_kernel_call(m_brgKernels[kernel_idx].get(), brgemmCtx, input_0, input_1, input_2, output_0);
+        h->add(output_0, brgemmCtx.N * io_data_size.back());
+        h->add(input_1, brgemmCtx.N * io_data_size[1]);
+        if (m_with_scratch && m_with_comp)
+            h->add(input_2, brgemmCtx.N * io_data_size[2]);
```

Comment:
BTW we can improve more.
Just imagine that we have only one full block (we don't have loop by N and don't have the tail). Seems like we can skip ptr increment as well.
Probably we don't need to do it now. I hope we will do it in bounds of the task with moving of loops from kernel to IR.
This comment is just thoughts out loud 🙂

Reply:
You are right. I improved the logic, thanks

Reply:
I meant to improve the logic in the near future. But you have already improved. Great! Thanks!